Userland/Libraries/LibGfx/ImageFormats/GIFWriter.cpp

```diff
@@ -127,6 +127,47 @@ ErrorOr<void> write_trailer(Stream& stream)
     return {};
 }
 
+class GIFAnimationWriter : public AnimationWriter {
+public:
+    GIFAnimationWriter(SeekableStream& stream)
+        : m_stream(stream)
+        , m_bit_stream(MaybeOwned(stream))
+    {
+    }
+
+    virtual ErrorOr<void> add_frame(Bitmap&, int, IntPoint) override;
+
+private:
+    SeekableStream& m_stream;
+    BigEndianOutputBitStream m_bit_stream;
```

Comment:
Individual frames are byte-aligned, right? If so, I'd create the bit stream in add_frame() each time, to make clear that frames always are on byte boundaries, and so that readers don't have to worry about sub-bit state vs seeks, etc

Reply:
That make sense, I put it here out of laziness